### DISCUSSION TITLE
Undici throwing error when parsing empty response with application/json content type

### DISCUSSION THREAD

**Discussion by cill-i-am** [nodejs/undici#3348] [Q&A] [answered]
Hey! So possibly dumb question. Undici is throwing an error when it tries to parse the response of a PUT request in our integration environment. The response body is empty & the response headers have `application/json` as a content type & content length of 0. It works as expected against the service when running in docker, but that doesn't append the content type. I'm just curious if this is expected behaviour?

```TS
ERROR Unexpected end of JSON input
│ SyntaxError: Unexpected end of JSON input
│     at JSON.parse (<anonymous>)
│     at parseJSONFromBytes (node:internal/deps/undici/undici:5329:19)
│     at successSteps (node:internal/deps/undici/undici:5300:27)
│     at fullyReadBody (node:internal/deps/undici/undici:1447:9)
│     at processTicksAndRejections (node:internal/process/task_queues:95:5)
│     at specConsumeBody (node:internal/deps/undici/undici:5309:7)
│     at Module.tsRestFetchApi 
```

I'm mostly looking for confirmation so that I can advocate for removing the content type from the empty response bodies. Just don't want to lead anyone in the wrong direction if not.

**Comment by mcollina** [comment, marked as answer]
A JSON is at least 2 bytes according to RFC 8259. The behavior of undici is correct.

**Reply by cill-i-am** [reply]
Thought as much. Thanks for the clarification 👍
